From 0114de4a06707ab9bc1805744a5ed741f00eb269 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Wed, 7 Mar 2012 08:00:26 +0100
Subject: [PATCH 12/12] wakeup on migration

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1331107226-21901-13-git-send-email-kraxel@redhat.com>
Patchwork-id: 38361
O-Subject: [RHEL-6.3 qemu-kvm PATCH v3 12/12] wakeup on migration
Bugzilla: 766303
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Gleb Natapov <gleb@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

Wakeup the guest when the live part of the migation is finished.
This avoids being in suspended state on migration, so we don't
have to save the is_suspended bit.

Temporary stop-gap for 6.3, will hash out proper suspend+migration
handling for upstream & 6.4, but that will not make the 6.3 deadline.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 migration.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 migration.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/migration.c b/migration.c
index fc17c57..fcaf7bc 100644
--- a/migration.c
+++ b/migration.c
@@ -399,6 +399,7 @@ void migrate_fd_put_ready(void *opaque)
         int old_vm_running = runstate_is_running();
 
         DPRINTF("done iterating\n");
+        qemu_system_wakeup_request(QEMU_WAKEUP_REASON_OTHER);
         vm_stop_force_state(RUN_STATE_FINISH_MIGRATE);
 
         qemu_aio_flush();
-- 
1.7.7.6

